lib/client/rpc/proposals.ts

```diff
@@ -242,7 +251,184 @@ export class FutarchyRPCProposalsClient implements FutarchyProposalsClient {
+            vaultAccount.conditionalOnFinalizeTokenMint,
+            this.rpcProvider.publicKey,
+            true
+          ),
+          userConditionalOnRevertTokenAccount: getAssociatedTokenAddressSync(
+            vaultAccount.conditionalOnRevertTokenMint,
+            this.rpcProvider.publicKey,
+            true
+          ),
+          userUnderlyingTokenAccount: getAssociatedTokenAddressSync(
+            vaultAccount.underlyingTokenMint,
+            this.rpcProvider.publicKey,
+            true
+          ),
+          vaultUnderlyingTokenAccount: vaultAccount.underlyingTokenAccount,
+          conditionalOnFinalizeTokenMint:
+            vaultAccount.conditionalOnFinalizeTokenMint,
+          conditionalOnRevertTokenMint:
+            vaultAccount.conditionalOnRevertTokenMint,
+        })
+        .instruction(),
+    ];
+    return ixs
+  }
+
+  async createProposal(daoAggregate: DaoAggregate, user: PublicKey) {
+    const proposalKP = Keypair.generate();
+    const proposal = proposalKP.publicKey;
+
+    console.log(proposal)
+    const currentDao = daoAggregate.daos.filter((dao) => dao.protocol.autocrat.programId.toString() == this.autocratClient.autocrat.programId.toString())[1]
+    if (!currentDao) return
+
+    console.log(currentDao.publicKey.toString())
+    const autocrat = this.autocratClient.autocrat
+    const vaultClient = this.autocratClient.vaultClient
+    const ammClient = this.autocratClient.ammClient
+    const treasury = currentDao.daoAccount.treasury
+
+    const usdcMint = currentDao.daoAccount.usdcMint;
+    const tokenMint = currentDao.daoAccount.tokenMint
+    if (!tokenMint) return
+    const vaultProgramId = vaultClient.vaultProgram.programId
+
+    console.log(usdcMint.toString(), tokenMint.toString())
+    const [baseVault] = getVaultAddr(vaultProgramId, treasury, tokenMint, proposal);
+    const [quoteVault] = getVaultAddr(vaultProgramId, treasury, usdcMint, proposal);
+
+    const [passBase] = getVaultFinalizeMintAddr(vaultProgramId, baseVault);
+    const [passQuote] = getVaultFinalizeMintAddr(vaultProgramId, quoteVault);
+    const [passAmm] = getAmmAddr(ammClient.program.programId, passBase, passQuote, proposal);
+
+    const [failBase] = getVaultRevertMintAddr(vaultProgramId, baseVault);
+    const [failQuote] = getVaultRevertMintAddr(vaultProgramId, quoteVault);
+    const [failAmm] = getAmmAddr(ammClient.program.programId, failBase, failQuote, proposal);
+
+    const daoAccount = await autocrat.account.dao.fetch(currentDao.publicKey)
+
+    const baseTokensToLP: BN = daoAccount.minBaseFutarchicLiquidity
+    const quoteTokensToLP: BN = daoAccount.minQuoteFutarchicLiquidity
+
+
+    console.log(baseTokensToLP.toNumber(), quoteTokensToLP.toNumber())
+    const baseBalance = await this.getUserTokenBalance(tokenMint)
+    const quoteBalance = await this.getUserTokenBalance(usdcMint)
+    console.log(baseBalance, quoteBalance)
+    const memoIx = this.createMemoInstruction()
```

Comment:
If you do want to shape this out, I recommend reviewing https://github.com/metaDAOproject/futarchy/blob/98e85dd82d84e889470488cd290e005cb863f39d/scripts/initializeProposal.ts#L89

You can also build out:

- `transferFromTreasury` https://github.com/metaDAOproject/futarchy/blob/98e85dd82d84e889470488cd290e005cb863f39d/scripts/initializeProposal.ts#L26
- `burnTreasury` https://github.com/metaDAOproject/futarchy/blob/98e85dd82d84e889470488cd290e005cb863f39d/scripts/initializeProposal.ts#L114

There may be more we're keen to implement, but for now this is what I've got.